### PR TITLE
Gan blog post

### PR DESCRIPTION
**Pull request by skaae**

### CODE REVIEW

**Comment by skaae** [comment]
If the blog does not compile it might be my not-ascii compatible name in authors.yaml

File: blog/_posts/2015-11-13-gan.md [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+
+
+<p align='center'><img width="75%" src="https://raw.githubusercontent.com/torch/torch.github.io/master/blog/_posts/images/model.png"></p>
+
+For this task, we employ a [Generative Adversarial Network](http://arxiv.org/abs/1406.2661) (GAN) [1].
+A GAN consists of two components; a *generator* which converts random noise into images and a *discriminator* which tries to distinguish between generated and real images. Here, 'real' means that the image came from our training set of images in contrast to the generated fakes.
+
+To train the model we let the discriminator and generator play a game against each other. We first show the discriminator a mixed batch of real images from our training set and of fake images generated by the generator. We then simultaneously optimize the discriminator to answer NO to fake images and YES to real images and optimize the generator to fool the discriminator into believing that the fake images were real. This corresponds to minimizing the classification error wrt. to the discriminator and maximizing it wrt. the generator.
```
Comment (soumith) [inline blog/_posts/2015-11-13-gan.md]: wrt. to the -> wrt. the

File: blog/_posts/2015-11-13-gan.md [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+
+### Balancing the GAN game
+In principle, the GAN optimization game is simple. We use binary cross entropy to optimize the parameters in the discriminator. Afterwards we use binary cross entropy to optimize the generator to fool the discriminator. That said, you often find yourself left with not very convincing outputs from generator:
+
+
+<p align='center'><img src="https://raw.githubusercontent.com/torch/torch.github.io/master/blog/_posts/images/bad_examples.png"></p>
+
+This gibberish is typical for a generator trained without proper care!
+
+A couple of tricks are necessary for to facilitate training: First of, we need to make sure that that neither the generator nor the discriminator becomes too good compared to the other. If the discriminator 'wins' and classifies all images correctly, the error signal will be poor and the generator will not be able to learn from it. Conversely, if we allow the generator to win, it is usually exploiting a non-meaningful weakness in the discriminator (e.g. by coloring the entire image blue) which is not desirable.
```
Comment (soumith) [inline blog/_posts/2015-11-13-gan.md]: First of -> First off

Comment (soumith) [inline blog/_posts/2015-11-13-gan.md]: becomes too good -> becomes too strong OR becomes much better

File: blog/_posts/2015-11-13-gan.md [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+
+ * Batch normalization speeds up training a lot when used in the generator. Using batch normalization in the discriminator is dangerous as the discriminator becomes too powerful.
+
+ * Plenty of dropout is needed in the discriminator to avoid oscillating behavior caused by the generator exploiting a weakness of the discriminator. Dropout can also be used in the generator.
+
+ * It may be beneficial to limit the capacity of the discriminator. This is done by decreasing its number of features such that the generator contains more parameters.
+
+
+### Generating faces
+We train our GAN using aligned and cropped images from the [Labeled faces in the wild](http://vis-www.cs.umass.edu/lfw/) dataset. After a around 5 epochs (around 30 minutes on a GPU) you should start to see some spooky faces (left).
```
Comment (soumith) [inline blog/_posts/2015-11-13-gan.md]: After a around -> After at around

File: blog/_posts/2015-11-13-gan.md [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+Which after 100 epochs will look more pleasant (right).
```
Comment (soumith) [inline blog/_posts/2015-11-13-gan.md]: Which after 100 -> And after 100

**Pull request merged**